docs: SNS and SQS FIFO (#4024)
aws/copilot-cli: site/content/docs/include/publish.en.md

<span class="parent-field">publish.topics.</span><a id="publish-topics-topic" href="#publish-topics-topic" class="field">`topic`</a> <span class="type">Map</span>  
Holds configuration for a single SNS topic.

> <span class="parent-field">publish.topics.topic.</span><a id="publish-topics-topic-topic-name" href="#publish-topics-topic-topic-name" class="field">`name`</a> <span class="type">String</span>  

Reviewer: ```suggestion
<span class="parent-field">publish.topics.topic.</span><a id="publish-topics-topic-name" href="#publish-topics-topic-topic-name" class="field">`name`</a> <span class="type">String</span>  
```

Required. The name of the SNS topic. Must contain only upper and lowercase letters, numbers, hyphens, and underscores.

<span class="parent-field">publish.topics.topic.</span><a id="publish-topics-topic-fifo" href="#publish-topics-topic-fifo" class="field">`fifo`</a> <span class="type">Boolean or Map</span>  
FIFO (first in, first out) SNS topic configuration.  

> If you specify `true`, Copilot will create the topic with FIFO capabilities.

Reviewer: ```suggestion
If you specify `true`, Copilot will create the topic with FIFO ordering.
```

```yaml
publish:
  topics:
    - name: mytopic
      fifo: true
```

> Alternatively, you can also configure the advanced SNS FIFO Topic configurations.

Reviewer: ```suggestion
Alternatively, you can also configure advanced SNS FIFO topic settings.
```

<span class="parent-field">publish.topics.topic.</span><a id="topic-name" href="#topic-name" class="field">`name`</a> <span class="type">String</span>  
Required. The name of the SNS topic. Must contain only upper and lowercase letters, numbers, hyphens, and underscores.

> <span class="parent-field">publish.topics.topic.</span><a id="fifo" href="#fifi" class="field">`fifo`</a> <span class="type">Boolean or Map</span>  

Reviewer 2: ```suggestion
<span class="parent-field">publish.topics.topic.</span><a id="publish-topics-topic-fifo" href="#publish-topics-topic-fifo" class="field">`fifo`</a> <span class="type">Boolean or Map</span>  
```
would you mind fixing this for the other places including https://github.com/aws/copilot-cli/pull/4024/files#diff-d0eb80518ed6d9a2e69c9f2a9cbe2b53a457323cc55296d23d14f8a59d969972R26 too?

Basically we need to ensure each item gets its own unique id and href.

> The FIFO SNS Topics configuration.   

Reviewer 3: ```suggestion
FIFO (first in, first out) SNS topic configuration.   
```

> If you specify true, Copilot will create SNS FIFO Topic with the given name and default FIFO settings.

Reviewer 3: ```suggestion
If you specify `true`, Copilot will create the topic with FIFO capabilities.
```

```yaml
publish:
  topics:
    - name: mytopic
      fifo: true
```

Alternatively, you can also configure the advanced SNS FIFO Topic configurations.
```yaml
publish:
  topics:
    - name: mytopic
      fifo:
        content_based_deduplication: true
```

<span class="parent-field">publish.topics.fifo</span><a id="ContentBasedDeduplication" href="#fifi" class="field">`content_based_deduplication`</a> <span class="type">Boolean</span>   

> content-based deduplication for FIFO topics

Reviewer 3: ```suggestion
If the message body is guaranteed to be unique for each published message, you can enable content-based deduplication for the SNS FIFO topic.
```
